test/integ/test_cursor.py

```diff
@@ -1331,11 +1337,20 @@ def test_resultbatch_lazy_fetching_and_schemas(conn_cnx, result_format, patch_pa
                 assert patched_download.call_count == 0
                 assert len(result_batches) > 5
                 assert result_batches[0]._local  # Sanity check first chunk being local
-                cur.fetchone()  # Trigger pre-fetching
-                # While the first chunk is local we still call _download on it, which
-                #  short circuits and just parses (for JSON batches) and then return an
-                #  an iterator through that data, so we expect the call count to be 5
-                assert patched_download.call_count == 5
+                first_output = cur.fetchone()
+                second_output = cur.fetchone()  # Trigger pre-fetching
+                assert first_output == (0,)
+                assert second_output == (1,)
+
+                # delay so that the thread pool downloads for all submitted chunks
+                time.sleep(1)
```

Comment:
I added this delay because the TPE needs time to call `_create_iter` on each batch that is in the pool. 
To avoid this I could mock the ThreadPoolExecutor object and just have my mock class call `_create_iter` right away. Let me know what you think and I can implement it.

Reply:
Usually the best way to test for something that might take up to `x` seconds is to check in a while loop where we allow failures, but if the condition is true then we short circuit, where `x` is what you'd deem unacceptable performance.

Now in this case this might not make much sense (for 1 second). Just keep this approach in mind in case you'd need to write a test that depends on some network activity. I'd say that mocking would take too much effort in this case.

Reply:
Thanks for letting me know 👍 . Since I just changed the iterator to pre-fetch though, it turns out I won't need to add a delay at all (the test remains unchanged). Good to know in case I need to write a test like this again.